homeassistant/components/tado/device_tracker.py

```diff
@@ -41,50 +52,99 @@ async def async_add_tracked_entities(
             continue
-        if device.get("location") and device["location"]["atHome"]:
-            _LOGGER.debug("Adding Tado device %s", device["name"])
-            new_tracked.append(
-                TadoDeviceTrackerEntity(str(device["id"]), device["name"])
-            )
-            tracked.add(device["id"])
 
-    _LOGGER.debug(
-        "Tado presence query successful, %d device(s) at home",
-        len(new_tracked),
-    )
-    async_add_entities(new_tracked)
+        _LOGGER.debug(
+            "Adding Tado device %s with deviceID %s", device["name"], device["id"]
+        )
+        new_tracked.append(
+            TadoDeviceTrackerEntity(device["id"], device["name"], tado_device)
+        )
+        tracked.add(device["id"])
+
+    async_add_entities(new_tracked, True)
 
 
 class TadoDeviceTrackerEntity(ScannerEntity):
     """A Tado Device Tracker entity."""
 
-    def __init__(self, device_id, device_name) -> None:
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        tado_device: Any,
+    ) -> None:
         """Initialize a Tado Device Tracker entity."""
         super().__init__()
-        self.device_id = device_id
-        self.device_name = device_name
+        self._device_id = device_id
+        self._device_name = device_name
+        self._tado_device = tado_device
+        self._active = False
+
+    @callback
+    async def async_update_state(self) -> None:
+        """Update the Tado device."""
+        _LOGGER.debug(
+            "Updating Tado device %s (ID: %s) device state",
+            self._device_name,
+            self._device_id,
+        )
+        devices = await self.hass.async_add_executor_job(
+            self._tado_device["data"].get_me
+        )
+        for device in devices["mobileDevices"]:
+            if device["id"] != self._device_id:
+                continue
+            self._active = False
+            if device.get("location") is not None and device["location"]["atHome"]:
+                _LOGGER.debug("Tado device %s is at home", device["name"])
+                self._active = True
+
+    @callback
+    async def async_on_demand_update(self) -> None:
+        """Update state on demand."""
+        await self.async_update_state()
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Register state update callback."""
+        _LOGGER.debug("Registering Tado device tracker entity")
+        await self.async_update_state()
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                "{DOMAIN}-update-{self.config_entry.entry_id}",
+                self.async_on_demand_update,
+            )
+        )
+
+        async def update(event_time: datetime) -> None:
+            """Update state."""
+            await self.async_on_demand_update()
+
+        self.async_on_remove(
+            async_track_time_interval(
```

Comment:
Schedule the update in the connector and send the updated devices as a parameter over the dispatch.

Reply:
Why would this be preferred? I've inspired myself from other integrations handling an `async_track_time_interval` this way. Once the `DataUpdateCoordinator` is in place this will be structurally fixed, anyhow. :)

Reply:
It will be more similar to what is already there today with the connector and what will be there afterwards with the DataUpdateCoordinator.